config.go

```diff
@@ -1032,26 +1037,41 @@ func (c ConfigTLS) LoadTLS(server string) (*tls.Config, error) {
 		InsecureSkipVerify: True(c.SkipVerify),
 	}
 
+	// Generate a hash of the contents of all certificates.
+	// This will be used to check for changes in the config or
+	// the cert contents, e.g. certificates getting rotated.
+	hash := sha256.New()
+
 	// Root CA (optional)
 	if c.CA != "" {
-		caCert, err := ioutil.ReadFile(c.CA)
+		f, err := os.Open(c.CA)
 		if err != nil {
-			return nil, err
+			return nil, "", err
 		}
+
+		data := io.TeeReader(f, hash)
+		caCert := new(bytes.Buffer)
+		caCert.ReadFrom(data)
```

Comment:
You sure these changes are needed? I think you can do:
```
hash.Write(caCert)

for _, bytes := range cert.Certificate {
	hash.Write(bytes)
}
```

Reply:
Yup.